notebooks/working/ho_deconfounder_on_sim_data.py

```diff
@@ -0,0 +1,430 @@
+mnl_names["total_travel_time"] = ['Travel Time, units:min (Drive Alone)',
+                                  'Travel Time, units:min (SharedRide-2)',
+                                  'Travel Time, units:min (SharedRide-3+)',
+                                  'Travel Time, units:min (All Transit Modes)']
+
+mnl_specification["total_travel_cost"] = [1, 2, 3, [4, 5, 6]]
+mnl_names["total_travel_cost"] = ['Travel Cost, units:$ (Drive Alone)',
+                                  'Travel Cost, units:$ (SharedRide-2)',
+                                  'Travel Cost, units:$ (SharedRide-3+)',
+                                  'Travel Cost, units:$ (All Transit Modes)']
+
+
+mnl_specification["cars_per_licensed_drivers"] = [[1, 2, 3]]
+mnl_names["cars_per_licensed_drivers"] = ["Autos per licensed drivers (All Auto Modes)"]
+
+mnl_specification["total_travel_distance"] = [1, 2, 3, 7, 8]
+mnl_names["total_travel_distance"] = ['Travel Distance, units:mi (Drive Alone)',
+                                      'Travel Distance, units:mi (SharedRide-2)',
+                                      'Travel Distance, units:mi (SharedRide-3+)',
+                                      'Travel Distance, units:mi (Walk)',
+                                      'Travel Distance, units:mi (Bike)']
+
+mnl_specification["cross_bay"] = [[2, 3]]
+mnl_names["cross_bay"] = ["Cross-Bay Tour (Shared Ride 2 & 3+)"]
+
+mnl_specification["household_size"] = [[2, 3]]
+mnl_names["household_size"] = ['Household Size (Shared Ride 2 & 3+)']
+
+mnl_specification["num_kids"] = [[2, 3]]
+mnl_names["num_kids"] = ["Number of Kids in Household (Shared Ride 2 & 3+)"]
+# -
+
+# # Recovering the confounder: The deconfounder algorithm
+
+# ## First approach: using all the variables in the factor model
+#
+# Using this approach, we use all the variables in each mode's utility specification to recover the confounder, in line with Blei et al.'s approach in their paper.
+
+# +
+## Return the variables of each utility specification
+
+spec_dic = specifications(mnl_specification=mnl_specification, num_modes=8)
+spec_dic
+# -
+
+confounder_vectors = []
+holdout_dfs = []
+masks_df = []
+rows_df =[]
+latent_dim = 1
+for i in data['mode_id'].unique():
+
+    data_mode_i = data[data['mode_id']==i]
+    # standardize the data for PPCA
+    print("Analysis for mode: ", i)
+    print("-------------------------------------------------------------------------------------------")
+    X_columns = spec_dic[i]
+    
+    if 'intercept' in X_columns:
+        X_columns.remove('intercept')
+        
+    if 'total_travel_distance' in X_columns:
+        X_columns.remove('total_travel_distance')
+        
+    print(i, X_columns)
+    
+    X = np.array((data_mode_i[X_columns] - data_mode_i[X_columns].mean())/data_mode_i[X_columns].std())
+    
+    confounders, holdouts, holdoutmasks, holdoutrow= confounder_ppca(holdout_portion=0.2, X=X, latent_dim=latent_dim)
```

Comment:
There's a bunch of plots and info about inferred axes and standard deviations that print here.
Readers currently have no explanation of what these plots are or what the printouts mean.
Can you remove them or explain them in detail?